### ISSUE TITLE
Converting a self-referencing Reusable block to regular blocks adds a new copy of the original Reusable block

### ISSUE THREAD

**Issue by dcalhoun** [WordPress/gutenberg#37204]
## Description
When converting a Reusable block to "regular" blocks, the original Reusable block is not removed from the block editor canvas. Instead of being replaced by the new regular blocks, the regular blocks are inserted alongside the original Reusable block.

## Step-by-step reproduction instructions
1. Add a Reusable block to the block editor canvas. 
2. Tap the Reusable block. 
3. Tap _Convert to regular blocks_. 

## Expected behaviour
The post contains only a set of regular blocks — a single copy of the content comprising the original Reusable block. 

## Actual behaviour
The post contains both a set of regular blocks and the original Reusable block, which is duplicative.

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/438664/145101948-e2aee3ce-73f4-4f6e-b671-dba379968961.MP4

## WordPress information
- WordPress version: WordPress.com
- Gutenberg version: v12.0.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? No, Geologist. 

## Device information
- Device: iPhone 13 mini
- Operating system: iOS 15.1.1
- WordPress app version: v18.8

**Comment by hypest** [comment]
Thanks for the report @dcalhoun! I've marked this one as "high priority" as it's a bad broken window for the editing experience.

Can you try diving in for a fix David? Thanks!

**Comment by dcalhoun** [comment]
@hypest after researching this further, I believe this may be working as intended. When I found this "bug" originally, I happened to be using a _self-referencing_ Reusable block. When one converts a self-referencing Reusable block, it inserts a copy of itself. (I feel a bit silly realizing this mistake now. 😆) 

Albeit a bit confusing, this behavior makes total sense. The web editor functions in this exact same way. When a Reusable block references itself an inline warning is displayed stating that it is not supported, so it is unlikely a user finds themself in the scenario of converting self-referencing blocks. 

I am closing this issue, but let me know if you believe we should reopen. Thanks! 🙇🏻
